testing/ef_tests/src/cases/fork_choice.rs

```diff
@@ -380,16 +404,72 @@ impl<E: EthSpec> Tester<E> {
             .unwrap();
     }
 
-    pub fn process_block(&self, block: SignedBeaconBlock<E>, valid: bool) -> Result<(), Error> {
+    pub fn process_block(
+        &self,
+        block: SignedBeaconBlock<E>,
+        blobs: Option<BlobsList<E>>,
+        kzg_proofs: Option<Vec<KzgProof>>,
+        valid: bool,
+    ) -> Result<(), Error> {
         let block_root = block.canonical_root();
+
+        // Convert blobs and kzg_proofs into sidecars, then plumb them into the availability tracker
+        if let Some(blobs) = blobs.clone() {
+            let proofs = kzg_proofs.unwrap();
+            let commitments = block
+                .message()
+                .body()
+                .blob_kzg_commitments()
+                .unwrap()
+                .clone();
+
+            // Zipping will stop when any of the zipped lists runs out, which is what we want. Some
+            // of the tests don't provide enough proofs/blobs, and should fail the availability
+            // check.
+            for (i, ((blob, kzg_proof), kzg_commitment)) in blobs
+                .into_iter()
+                .zip(proofs)
+                .zip(commitments.into_iter())
+                .enumerate()
+            {
+                let signed_sidecar = SignedBlobSidecar {
+                    message: Arc::new(BlobSidecar {
+                        block_root,
+                        index: i as u64,
+                        slot: block.slot(),
+                        block_parent_root: block.parent_root(),
+                        proposer_index: block.message().proposer_index(),
+                        blob,
+                        kzg_commitment,
+                        kzg_proof,
+                    }),
+                    signature: Signature::empty(),
+                    _phantom: Default::default(),
+                };
+                let result = self.block_on_dangerous(
+                    self.harness
+                        .chain
+                        .check_gossip_blob_availability_and_import(
+                            GossipVerifiedBlob::__assumed_valid(signed_sidecar),
+                        ),
+                )?;
+                if valid {
+                    assert!(result.is_ok());
+                }
```

Comment:
do we want to check for the invalid case here, i.e. `assert_eq!(valid, result.is_ok())`?

Reply:
no, because some blobs from the block may be valid even if the whole block with blobs is invalid

Reply:
ah got it, thanks!

Reply:
letss merge 🚀